jsonschema2pojo-maven-plugin/src/main/java/org/jsonschema2pojo/maven/Jsonschema2PojoMojo.java

```diff
@@ -1114,6 +1118,40 @@ public boolean isIncludeAdditionalProperties() {
     @Override
     public boolean isIncludeSetters() { return includeSetters; }
 
+    private void setTargetVersion() {
+        if (isNotBlank(this.targetVersion)) {
+            return;
+        }
+
+        if (project.getProperties() != null && project.getProperties().containsKey("maven.compiler.source")) {
+            this.targetVersion = project.getProperties().get("maven.compiler.source").toString();
+            return;
+        }
+
+        if (project.getProperties() != null && project.getProperties().containsKey("maven.compiler.release")) {
+            this.targetVersion = project.getProperties().get("maven.compiler.release").toString();
+            return;
+        }
+
+        for (Plugin p : (List<Plugin>) project.getBuildPlugins()) {
+            if (p.getKey().equals("org.apache.maven.plugins:maven-compiler-plugin") && p.getConfiguration() instanceof Xpp3Dom) {
+                final Xpp3Dom compilerSourceConfig = ((Xpp3Dom) p.getConfiguration()).getChild("source");
```

Comment:
should we prefer the `target` property first?
i.e. same as in the property `defaultValue`

Reply:
and yep, I need to update `defaultValue = "${maven.compiler.target}"`. I had missed this after bringing in the recent changes to move to the `Parameter` annotation. Thanks for mentioning!

Reply:
i'll trust you to make the right choice between target or source, if we are generating code then yeah source seems better

my point was only that it seemed inconsistent to default to `target` but not use `target` from `maven-compiler-plugin` here as well :sweat_smile:
if the checking order is "source, release" in both then that seems fine :+1: